model/import/StaticAssemblerService.php

```diff
@@ -0,0 +1,105 @@
+<?php
+/**  
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ * 
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * 
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ * 
+ * Copyright (c) 2013 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ * 
+ */
+namespace oat\taoDeliveryRdf\model\import;
+
+use common_Exception;
+use oat\generis\model\OntologyAwareTrait;
+use oat\taoQtiTest\models\PhpCodeCompilationDataService;
+use oat\taoQtiTest\models\XmlCompilationDataService;
+use tao_helpers_File;
+use tao_models_classes_service_ServiceCall;
+use tao_models_classes_service_StorageDirectory;
+use ZipArchive;
+
+/**
+ * AssemblerService Class.
+ *
+ * Im- and export a compiled delivery 
+ *
+ * @access public
+ * @author Joel Bout, <joel@taotesting.com>
+ * @package taoDelivery
+ */
+class StaticAssemblerService extends AssemblerService
```

Comment:
Maybe instead of inheritance it would be better to extract related functionality into separate class(es) and based on input parameters return content as is or transform it.

Reply:
I guess no. Could you explain to me please, why it would be better? AssemblerService is a way how to work with deliveries (which content of the delivery we can use in the environment), but it doesn't mean that we can't use another type in different places. So in the case when I configure AssemblerService parameter (for example contentType with 3 options: phpcode, phpserialization, xmlcompilation) which also can be mixed, I have to redeclare all the AssemblerService service, and I don't see why it would be easier to implement or to support.